File: 2742. Painting the Walls/Solution.py
class Solution:
    def paintWalls(self, cost: List[int], time: List[int]) -> int:
        n = len(cost)
        dp = [1000000007 for i in range(n + 1)]
        dp[0] = 0
        for c, t in zip(cost, time):
            for i in range(n, -1, -1):
                dp[min(i + t + 1, n)] = min(dp[min(i + t + 1, n)], dp[i] + c)
        return dp[n]

File: 2742. Painting the Walls/Solution.cpp
class Solution {
public:
    int paintWalls(vector<int>& cost, vector<int>& time) {
        int n = cost.size();
        vector<int> dp(n + 1, 1000000007);
        dp[0] = 0;
        for (int i = 0; i < n; i++) {
            int c = cost[i], t = time[i];
            for (int j = n; j >= 0; j--) {
                dp[min(j + t + 1, n)] = min(dp[min(j + t + 1, n)], dp[j] + c);
            }
        }
        return dp[n];
    }
};
